conans/client/source.py

```diff
@@ -76,17 +77,24 @@ def remove_source(raise_error=True):
             raise ConanException(e)
 
 
+def _handle_scm(scm, conanfile, src_folder, output):
+    if scm.is_source_folder():
+        output.warn("SCM not fetching sources. Copying sources from: %s" % scm.url)
+        # Maybe this can be avoided and just point source_folder to user folder?
+        shutil.copytree(scm.url, src_folder, symlinks=True)
+        return
+    elif scm.is_git():
+        output.info("Getting sources from: %s - %s" % (scm.url, scm.checkout))
+        conanfile.run('git clone "%s" "%s"' % (scm.url, src_folder))
+        conanfile.run('git checkout "%s"' % scm.checkout, cwd=src_folder)
```

Comment:
For better abstraction, the clone & checkout commands should be part of SCM class.